src/ingest-pipeline/airflow/dags/rebuild_multiple_metadata.py

```diff
@@ -46,18 +46,21 @@ def build_dataset_lists(**kwargs):
         pprint(kwargs["dag_run"].conf)
         for uuid in kwargs["dag_run"].conf["uuids"]:
             soft_data = get_soft_data(uuid, **kwargs)
+            ds_rslt = pythonop_get_dataset_state(
+                dataset_uuid_callable=lambda **kwargs: uuid, **kwargs
+            )
 
             # If we got nothing back from soft_data, then let's try to determine using entity_api
             if soft_data:
                 if soft_data.get("primary"):
-                    kwargs["dag_run"].conf["primary_datasets"].append(uuid)
+                    if ds_rslt["creation_action"] == "Multi-Assay Split":
+                        kwargs["dag_run"].conf["component_datasets"].append(uuid)
```

Comment:
Maybe you are missing the declaration of "component_datasets"? like the one you have for "primary_datasets" and "processed_datasets"

Reply:
Good catch @sunset666 - adjusted to include this